File: src/main/java/com/example/HealthAndFitnessPlatform/model/Favorite.java
package com.example.HealthAndFitnessPlatform.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "favoriteTBL")
@Data
@AllArgsConstructor
@RequiredArgsConstructor
public class Favorite {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "userId")
    private User user;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "recipeId")
    private Recipe recipe;

    @Column(name = "createdAt")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt = LocalDateTime.now();


    @PrePersist
    public void prePersist() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
    }
}

File: HealthAndFitnessPLatformNG/src/app/model/favorite.ts
export class Favorite {
     
               id?:number;
               userId:number;
               recipeId:number;
               createdAt?:string;

               constructor(id:number,userId:number,recipeId:number,createdAt:string){
                         this.id = id;
                         this.userId = userId;
                         this.recipeId = recipeId;
                         this.createdAt = createdAt;
               }
}